aiida/calculations/plugins/transfer.py

```diff
@@ -0,0 +1,179 @@
+            ' - If you want to retrieve use `retrieve: True`\n'
+            ' - If you want to just gather use `retrieve: False`\n'
+        )
+        return errmsg.format(instructions.uuid)
+    object_list = instructions.get_dict()['object_list']
+
+    source_uuids = [node.pk for node in source_nodes.values()]
+    source_pkids = [node.uuid for node in source_nodes.values()]
+    instruct_ids = [src_node_id for src_node_id, src_locpath, tgt_locpath in object_list]
+
+    for node_id in instruct_ids:
+        its_a_uuid = node_id in source_uuids
+        its_a_pkid = node_id in source_pkids
+        if not (its_a_uuid or its_a_pkid):
+            errmsg = 'node identified by `{}` in instructions has not been passed as an input.'
+            return errmsg.format(node_id)
+
+    for nodename, node in source_nodes.items():  # pylint: disable=unused-variable
+        its_a_uuid = node.uuid in instruct_ids
+        its_a_pkid = node.pk in instruct_ids
+        if not (its_a_uuid or its_a_pkid):
+            errmsg = 'input node provided (pk=`{}`, uuid=`{}`) is not used in instructions.'
+            return errmsg.format(node.pk, node.uuid)
+
+
+# Just keeping this as a reference of how to do single input validation
+#def validate_remote_path(target_path):  # pylint: disable=inconsistent-return-statements
+#    """target_path seems to be an empty list by default: ()"""
+#    if target_path:
+#        if os.path.isabs(target_path):
+#            return 'absolute path not good'
+
+
+class TransferCalculation(CalcJob):
+    """Implementation of CalcJob to add two numbers for testing and demonstration purposes."""
+
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+        spec.input('instructions', valid_type=orm.Dict)
+        spec.input_namespace('source_nodes', valid_type=(orm.FolderData, orm.RemoteData), dynamic=True, required=True)
+        spec.input('metadata.options.remote_path', valid_type=(str, type(None)), default=None)
+
+        # Keep a reference of how to use validators + eventually renaming the retrieved and
+        # the remote_folder
+        # spec.input('metadata.options.target_path', valid_type=str, required=False, validator=validate_remote_path)
+        # spec.output('local', valid_type=orm.FolderData, required=False)
+        # spec.output('remote', valid_type=orm.RemoteData, required=False)
+
+        # Code should only be required if going through a queue
+        spec.inputs['code'].required = False
+
+        # Retrieved node should only be generated if actually retrieving files
+        # (we need to check if it is possible to have this be consistent with the
+        # autogeneration of files when no script is ran)
+        spec.outputs[cls.link_label_retrieved].required = False
+
+        # Resources metadata should only be required if going through a queue
+        spec.inputs['metadata']['options']['resources'].required = False
```

Comment:
Ok, I just commented all these sets to `False` since these are anyways needed for the execmanager not to fail, so this way I am just removing checks but still having to provide the inputs which made no sense...